src/reference/asciidoc/codec.adoc

```diff
@@ -0,0 +1,122 @@
+[[codec]]
+=== Codec
+
+==== Introduction
+
+Spring Integration _version 4.2_ introduces the `Codec` abstraction.
+Codecs are used to encode/decode objects to/from byte[].
+They are an alternative to Java Serialization.
+One advantage is, typically, objects do not have to implement Serializable
+One implementation, using https://github.com/EsotericSoftware/kryo[Kryo] for serialization is provided but you
```

Comment:
No dot in the end of previous sentence.
For it looks better to make `Serializable` in code formatting.
The same for `byte[]` above.
The comma after ` for serialization` here.